include/fschuetz04/simcpp20/value_event.hpp

```diff
@@ -168,8 +168,6 @@ class value_event : public event<Time> {
 
     /// Destructor.
     ~data() override {
-      event<Time>::data::~data();
```

Comment:
This makes sense, as virtual base class destructors are called automatically anyway. Calling the destructor a second time here causes the segfault, as the coroutine handles are destroyed a second time. Removing this line fixes the segfault.